packages/collector/test/tracing/database/redis/test.js

```diff
@@ -21,748 +22,783 @@ const {
-              })
-            );
-          }));
-
-      it('must not trace get without waiting', () =>
-        controls
-          .sendRequest({
-            method: 'GET',
-            path: '/get-without-waiting',
-            qs: {
-              key: 'price'
-            }
-          })
-          .then(() =>
-            retry(() =>
-              agentControls.getSpans().then(spans => {
-                const writeEntrySpan = expectAtLeastOneMatching(spans, [
-                  span => expect(span.n).to.equal('node.http.server'),
-                  span => expect(span.data.http.method).to.equal('GET')
-                ]);
-
-                verifyHttpExit(spans, writeEntrySpan);
-              })
-            )
-          ));
-
-      it('must trace set without cb', () =>
-        controls
-          .sendRequest({
-            method: 'GET',
-            path: '/set-without-waiting',
-            qs: {
-              key: 'price',
-              value: 42
+          mochaSuiteFn(`redis@${redisVersion}`, function () {
+            globalAgent.setUpCleanUpHooks();
+            let controls;
+
+            before(async () => {
+              controls = new ProcessControls({
+                useGlobalAgent: true,
+                appPath:
+                  redisVersion === 'latest' ? path.join(__dirname, 'app.js') : path.join(__dirname, 'legacyApp.js'),
+                env: {
+                  REDIS_VERSION: redisVersion,
+                  REDIS_PKG: redisPkg,
+                  REDIS_CLUSTER: setupType === 'cluster'
+                }
+              });
+
+              await controls.startAndWaitForAgentConnection(5000, Date.now() + 1000 * 60 * 5);
+            });
+
+            beforeEach(async () => {
+              await agentControls.clearReceivedTraceData();
+            });
+
+            before(async () => {
+              await controls.sendRequest({
+                method: 'POST',
+                path: '/clearkeys'
+              });
+            });
+
+            after(async () => {
+              await controls.stop();
+            });
+
+            afterEach(async () => {
+              await controls.clearIpcMessages();
+            });
+
+            it('must trace set/get calls', () =>
+              controls
+                .sendRequest({
+                  method: 'POST',
+                  path: '/values',
+                  qs: {
+                    key: 'price',
+                    value: 42
+                  }
+                })
+                .then(() =>
+                  controls.sendRequest({
+                    method: 'GET',
+                    path: '/values',
+                    qs: {
+                      key: 'price'
+                    }
+                  })
+                )
+                .then(response => {
+                  expect(String(response)).to.equal('42');
+
+                  return retry(() =>
+                    agentControls.getSpans().then(spans => {
```

Comment:
Question: Should we consider adding a check for the count of the spans?

Reply:
Yeah lets add that 👍 Good catch